src/strategies/DSL/actions/limitSwapAction.ts

```diff
@@ -1,17 +1,59 @@
-import { LimitSwapActionArgs, PrimitiveArgs } from '@brinkninja/types'
+import { LimitSwapActionArgs, PrimitiveArgs, PrimitiveParamValue } from '@brinkninja/types'
+import { convertToX96HexPrice, toBigInt, toTokenArgs } from '@brink-sdk/internal';
+
+const FLAT_PRICE_CURVE_ADDRESS = '0xC509733B8dDdbab9369A96F6F216d6E59DB3900f';
+
+export class InvalidInputError extends Error {
+    constructor(message: string) {
+        super(message);
+        this.name = 'InvalidInputError';
+    }
+}
 
 function limitSwapAction ({
+  id,
   owner,
   tokenIn,
   tokenOut,
   tokenInAmount,
-  tokenOutAmount,
-  price
+  tokenOutAmount
 }: LimitSwapActionArgs): PrimitiveArgs[] {
+  if (tokenOutAmount === undefined) {
```

Comment:
should we check the other params as well?

there are tools we could use for this. @MatiasOS uses https://github.com/hapijs/joi for API request validation. could be something to consider for DSL

Reply:
i think it's the only one? the other ones are checked by the type system.

Reply:
@phanpak types aren't guaranteed at runtime though. devs can pass whatever JSON they want to DSL, it doesn't have to be type checked. could potentially be generated in any language and sent to our API.

We can address validation in another ticket though, I'll make one in linear